skbeam/core/correlation.py

```diff
@@ -405,3 +408,359 @@ def auto_corr_scat_factor(lags, beta, relaxation_rate, baseline=1):
+    frames.
+
+    The longest lag time computed is num_levels * num_bufs.
+    ** see comments on multi_tau_auto_corr
+
+    Parameters
+    ----------
+    labels : array
+        labeled array of the same shape as the image stack;
+        each ROI is represented by a distinct label (i.e., integer)
+    images : array
+        dimensions are: (rr, cc), iterable of 2D arrays
+    num_frames : int
+        number of images to use
+        default is number of images
+    num_bufs : int, must be even
+        maximum lag step to compute in each generation of
+        downsampling
+        default is number of images
+    num_levels : int, optional
+        how many generations of downsampling to perform, i.e.,
+        the depth of the binomial tree of averaged frames
+        default is one
+
+    Yields
+    ------
+    namedtuple
+        A `results` object is yielded after every image has been processed. This
+        `reults` object contains:
+        - the normalized correlation, `two_time`
+        - the times at which the correlation was computed, `lag_steps`
+        - and all of the internal state, `final_state`, which is a
+          `correlation_state` namedtuple
+
+    Returns
+    -------
+    two_time : array
+        matrix of two time correlation
+        shape (number of images, number of images, number of labels(ROI))
+
+    Notes
+    -----
+    The two-time correlation function is defined as
+
+    :math ::
+        C(q, t_1, t_2) = \frac{<I(q, t_1)I(q, t_2)>_pix }{<I(q, t_1)>_pix <I(q, t_2)>_pix}
+
+    Here, the ensemble averages are performed over many pixels of detector,
+    all having the same q value. The average time or age is equal to (t1+t2)/2,
+    measured by the distance along the t1 = t2 diagonal.
+    The time difference t = |t1 - t2|, with is distance from the t1 = t2
+    diagonal in the perpendicular direction.
+    In the equilibrium system, the two-time correlation functions depend only
+    on the time difference t, and hence the two-time correlation contour lines
+    are parallel.
+
+    References
+    ----------
+
+    .. [1] A. Fluerasu, A. Moussaid, A. Mandsen and A. Schofield,
+        "Slow dynamics and aging in collodial gels studied by x-ray photon
+         correlation spectroscopy," Phys. Rev. E., vol 76, p 010401(1-4), 2007.
+    """
+    if two_time_internal_state is None:
+        two_time_internal_state = _init_state_two_time(num_levels, num_bufs,
+                                                       labels, num_frames)
+    # create a shorthand reference to the results and state named tuple
+    s = two_time_internal_state
+
+    # generate a time frame for each level
+    time_ind = {key: [] for key in range(num_levels)}
+
+    for img in images:
+        s.cur[0] = (1 + s.cur[0]) % num_bufs  # increment buffer
+
+        s.count_level[0] = 1 + s.count_level[0]
+
+        # get the current image time
+        s = s._replace(current_img_time=(s.current_img_time + 1))
```

Comment:
In two time you need to track the current_img_time because it will use to find the t1 and t2 values for the two time correlation(two times). The final results store in the results two_time(t1, t2)

Check from line 636  ..., It shows how it use.